toolchain/check/import_cpp.cpp

```diff
@@ -310,13 +310,28 @@ static auto MakeIntType(Context& context, IntId size_id) -> TypeExpr {
   return ExprAsType(context, Parse::NodeId::None, type_inst_id);
 }
 
-// Maps a C++ type to a Carbon type. Currently only 32-bit `int` is supported.
+// Maps a C++ type to a Carbon type. Currently only 32-bit `int` and 16-bit
+// `short` are supported.
 // TODO: Support more types.
 static auto MapType(Context& context, clang::QualType type) -> TypeExpr {
   const auto* builtin_type = dyn_cast<clang::BuiltinType>(type);
-  if (builtin_type && builtin_type->getKind() == clang::BuiltinType::Int &&
-      context.ast_context().getTypeSize(type) == 32) {
-    return MakeIntType(context, context.ints().Add(32));
+  if (!builtin_type) {
+    return {.inst_id = SemIR::ErrorInst::TypeInstId,
+            .type_id = SemIR::ErrorInst::TypeId};
+  }
+  switch (builtin_type->getKind()) {
+    case clang::BuiltinType::Short:
+      if (context.ast_context().getTypeSize(type) == 16) {
+        return MakeIntType(context, context.ints().Add(16));
+      }
+      break;
+    case clang::BuiltinType::Int:
+      if (context.ast_context().getTypeSize(type) == 32) {
+        return MakeIntType(context, context.ints().Add(32));
+      }
+      break;
+    default:
+      break;
```

Comment:
Is this necessary? Do have an option to make the compiler complain if we don't have this?

Reply:
Yes, `-Wswitch` is part of `-Wall`.

(Aside: Clang is a bit unreliable with that warning though, so false negatives (no warning) doesn't mean it would not warn in the future when the enum changed in some way.)